mapper.go

```diff
@@ -58,7 +59,7 @@ func (m *metricMapper) initFromString(fileContents string) error {
 
 		switch state {
 		case SEARCHING:
-			if line == "" {
+			if line == "" || line[0] == COMMENT_CHAR {
```

Comment:
The way this loop is designed to change state on blank lines, this case becomes difficult to deal with:

```
foo
bar=1
# comment
ping
pong=1
```

I think I can factor out the blank line as the cue for the end of a metric definition.